lib/segment/src/index/sparse_index/sparse_vector_index.rs

```diff
@@ -26,28 +35,150 @@ impl<TInvertedIndex: InvertedIndex> SparseVectorIndex<TInvertedIndex> {
     pub fn search_sparse(
         &self,
-        query: SparseVector,
+        vectors: &[&QueryVector],
         top: usize,
         is_stopped: &AtomicBool,
-    ) -> Vec<ScoredPointOffset> {
-        let mut search_context = SearchContext::new(query, top, &self.inverted_index, is_stopped);
-        search_context.search()
+        with_filter: bool,
+        condition: impl Fn(PointOffsetType) -> bool,
+    ) -> OperationResult<Vec<Vec<ScoredPointOffset>>> {
+        let mut result = vec![];
+
+        for vector in vectors {
+            check_process_stopped(is_stopped)?;
+            // measure time according to filter
+            if with_filter {
+                let _timer = ScopeDurationMeasurer::new(&self.searches_telemetry.filtered_sparse);
+            } else {
+                let _timer = ScopeDurationMeasurer::new(&self.searches_telemetry.unfiltered_sparse);
+            }
+            let vector = match vector {
+                QueryVector::Nearest(vector) => vector,
+                QueryVector::Recommend(_) => {
+                    return Err(OperationError::ValidationError {
+                        description: "Cannot recommend sparse vectors".to_string(),
+                    })
+                }
+                QueryVector::Discovery(_) => {
+                    return Err(OperationError::ValidationError {
+                        description: "Cannot discovery sparse vectors".to_string(),
+                    })
+                }
+                QueryVector::Context(_) => {
+                    return Err(OperationError::ValidationError {
+                        description: "Cannot context query sparse vectors".to_string(),
+                    })
+                }
+            };
+            let sparse_vector: &SparseVector = vector.to_vec_ref().try_into()?;
+            let mut search_context = SearchContext::new(
+                sparse_vector.to_owned(),
+                top,
+                &self.inverted_index,
+                is_stopped,
+            );
+            let points = search_context.search(&condition);
+            result.push(points);
+        }
+
+        Ok(result)
     }
 }
 
-// TODO impl VectorIndex for SparseVectorIndex
+impl<TInvertedIndex: InvertedIndex> VectorIndex for SparseVectorIndex<TInvertedIndex> {
+    fn search(
+        &self,
+        vectors: &[&QueryVector],
+        filter: Option<&Filter>,
+        top: usize,
+        _params: Option<&SearchParams>,
+        is_stopped: &AtomicBool,
+    ) -> OperationResult<Vec<Vec<ScoredPointOffset>>> {
+        let id_tracker = self.id_tracker.borrow();
+        let vector_storage = self.vector_storage.borrow();
+        let deleted_point_bitslice = id_tracker.deleted_point_bitslice();
+        let deleted_vectors = vector_storage.deleted_vector_bitslice();
+        // filter for deleted points
+        let not_deleted_condition = |idx: PointOffsetType| -> bool {
+            !deleted_point_bitslice
```

Comment:
This is a copypaste from raw scorer. We have to do some abstraction with this check of deleted vector. Or mark as `TODO(sparse)` =)

Reply:
I decided to both reuse the implementation and add a TODO.
This is a bit of a hack but will make sure we don't forget to update the implementation.

A wider refactoring is necessary to remove the need for a dedicated implementation for `SparseRawScorer`.
cc @coszio :)